dot/parachain/availability-store/availability_store.go

```diff
@@ -4,61 +4,160 @@
+	"github.com/ChainSafe/gossamer/lib/trie"
+	"github.com/ChainSafe/gossamer/pkg/scale"
 )
 
 var logger = log.NewFromGlobal(log.AddContext("pkg", "parachain-availability-store"))
 
 const (
-	avaliableDataPrefix = "available"
+	availableDataPrefix = "available"
 	chunkPrefix         = "chunk"
 	metaPrefix          = "meta"
 	unfinalizedPrefix   = "unfinalized"
 	pruneByTimePrefix   = "prune_by_time"
+
+	tombstoneValue = " "
+
+	// Unavailable blocks are kept for 1 hour.
+	KEEP_UNAVAILABLE_FOR = time.Hour
+
+	// Finalized data is kept for 25 hours.
+	KEEP_FINALIZED_FOR = time.Hour * 25
+
+	// The pruning interval.
+	PRUNING_INTERVAL = time.Minute * 5
 )
 
+type BETimestamp uint64
+
+func (b BETimestamp) ToBytes() []byte {
+	res := make([]byte, 8)
+	binary.LittleEndian.PutUint64(res, uint64(b))
+	return res
+}
+
+type SubsystemClock struct{}
+
+func (sc *SubsystemClock) Now() BETimestamp {
+	return BETimestamp(time.Now().Unix())
+}
+
+// PruningConfig Struct holding pruning timing configuration.
+// The only purpose of this structure is to use different timing
+// configurations in production and in testing.
+type PruningConfig struct {
+	keepUnavailableFor time.Duration
+	keepFinalizedFor   time.Duration
+	pruningInterval    time.Duration
+}
+
+var DefaultPruningConfig = PruningConfig{
+	keepUnavailableFor: KEEP_UNAVAILABLE_FOR,
+	keepFinalizedFor:   KEEP_FINALIZED_FOR,
+	pruningInterval:    PRUNING_INTERVAL,
+}
+
 // AvailabilityStoreSubsystem is the struct that holds subsystem data for the availability store
 type AvailabilityStoreSubsystem struct {
 	SubSystemToOverseer chan<- any
 	OverseerToSubSystem <-chan any
 	availabilityStore   AvailabilityStore
-	//TODO: pruningConfig PruningConfig
-	//TODO: clock         Clock
+	pruningConfig       PruningConfig
+	clock               SubsystemClock
 	//TODO: metrics       Metrics
 }
 
 // AvailabilityStore is the struct that holds data for the availability store
 type AvailabilityStore struct {
-	availableTable database.Table
-	chunkTable     database.Table
-	metaTable      database.Table
-	//TODO: unfinalizedTable database.Table
-	//TODO: pruneByTimeTable database.Table
+	availableTable   database.Table
+	chunkTable       database.Table
+	metaTable        database.Table
+	unfinalizedTable database.Table
+	pruneByTimeTable database.Table
+}
+
+type AvailabilityStoreBatch struct {
+	availableBatch   database.Batch
+	chunkBatch       database.Batch
+	metaBatch        database.Batch
+	unfinalizedBatch database.Batch
+	pruneByTimeBatch database.Batch
+}
+
+func NewAvailabilityStoreBatch(as *AvailabilityStore) *AvailabilityStoreBatch {
+	return &AvailabilityStoreBatch{
+		availableBatch:   as.availableTable.NewBatch(),
+		chunkBatch:       as.chunkTable.NewBatch(),
+		metaBatch:        as.metaTable.NewBatch(),
+		unfinalizedBatch: as.unfinalizedTable.NewBatch(),
+		pruneByTimeBatch: as.pruneByTimeTable.NewBatch(),
+	}
+}
+
+func (asb *AvailabilityStoreBatch) Write() error {
```

Comment:
Since this function flushes all the batches, I would suggest to call it `Flush`
```suggestion
func (asb *AvailabilityStoreBatch) Flush() error {
```

Reply:
Flush does make more sense, I was using `Write` to conform to the naming that Parity is using in Polkadot, but this makes is confusing.  I've updated to Flush.

Reply:
@EclesioMeloJunior I've updated this.